ckanext/fjelltopp_security/plugin.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import ckan.plugins as plugins
+import ckan.plugins.toolkit as toolkit
+import ckanext.fjelltopp_security.actions as fjelltopp_security_actions
+
+log = logging.getLogger(__name__)
+
+
+class FjelltoppSecurityPlugin(plugins.SingletonPlugin):
+    plugins.implements(plugins.IConfigurer)
+    plugins.implements(plugins.IMiddleware, inherit=True)
+    plugins.implements(plugins.IActions)
+    # plugins.implements(plugins.IValidators)
+
+
+    # plugins.implements(plugins.IConfigurable)
+
+    # IConfigurer
+    def update_config(self, config_):
+        toolkit.add_template_directory(config_, "templates")
+        toolkit.add_public_directory(config_, "public")
+        toolkit.add_resource("assets", "fjelltopp-security")
+
+    # IValidators
+
+    # IActions
+    def get_actions(self):
+        return {
+            'user_update': fjelltopp_security_actions.secure_user_update,
+            'user_create': fjelltopp_security_actions.secure_user_create,
+            'group_update': fjelltopp_security_actions.secure_group_update,
+            'group_create': fjelltopp_security_actions.secure_group_create,
+            'organization_update': fjelltopp_security_actions.secure_organization_update,
+            'organization_create': fjelltopp_security_actions.secure_organization_create,
+        }
+
+    # IMiddleware
+    def make_middleware(self, app, config):
+        @app.after_request
+        def apply_owasp(response):
+            response.headers["Strict-Transport-Security"] = config.get(
+                "ckanext.fjelltopp_security.strict_transport_security",
+                "max-age=31536000; preload",
+            )
+            response.headers["X-Content-Type-Options"] = config.get(
+                "ckanext.fjelltopp_security.content_type_options", "nosniff"
+            )
+            response.headers["X-Permitted-Cross-Domain-Policies"] = config.get(
+                "ckanext.fjelltopp_security.cross_domain_policies",
+                "none",  # not sure about this one
+            )
+            response.headers["Referrer-Policy"] = config.get(
+                "ckanext.fjelltopp_security.referrer_policy",
+                "no-referrer-when-downgrade",  # this is default when not set
+            )
+            response.headers["Cache-Control"] = config.get(
+                "ckanext.fjelltopp_security.cache_control",
+                "",
+            )
+            response.headers["Cross-Origin-Opener-Policy"] = config.get(
+                "ckanext.fjelltopp_security.coop", "same-site"
+            )
+            response.headers["Cross-Origin-Embedder-Policy"] = config.get(
+                "ckanext.fjelltopp_security.coep", "unsafe-none"
+            )
+            response.headers["Cross-Origin-Resource-Policy"] = config.get(
+                "ckanext.fjelltopp_security.corp", "cross-origin"
+            )
```

Comment:
I'd prefer the full names over `coop`, `coep` and `corp` I think as it's fairly unclear when just viewing this from the settings file/PR